productinfo/gce/network_mapper.go

```diff
@@ -24,5 +25,19 @@ func newGceNetworkMapper() *GceNetworkMapper {
 
 // MapNetworkPerf maps the network performance of the gce instance to the category supported by telescopes
 func (nm *GceNetworkMapper) MapNetworkPerf(vm productinfo.VmInfo) (string, error) {
-	return ntwPerfMap[productinfo.NTW_MEDIUM][0], nil
+	for perfCat, strVals := range ntwPerfMap {
+		if contains(strVals, vm.NtwPerf) {
+			return perfCat, nil
+		}
+	}
+	return "", fmt.Errorf("could not determine network performance for: [%s]", vm.NtwPerf)
+}
+
+func contains(slice []string, val string) bool {
```

Comment:
please extract this method to the provider agnostic productinfo file (or find another solution to aviud implementing it for each provider)